### PR TITLE
updated readme

### PR DESCRIPTION
**Pull request by daiLlew**
### What

- Rewrote the `README` to hopefully include some useful information.
- Created an initial swagger spec for the API (covers endpoints `begin`, `commitManifest`and `publish`)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,33 +1,95 @@
...
+
+Below is a list of the available endpoints with a brief description of what they do. For more detail on the API you 
+can checkout the `swagger.yml` in the root of this project.
  
-By default the publisher will operate on temp directories. It prints out console messages about the configuration variables you can use to set up directories in production, or take a look at the `Configuration` class.
+| Endpoint               | Method    | Description                                                               |
+| ---------------------- | --------- | --------------------------------------------------------------------------| 
+| **/begin**             | **POST**  | Create a transation for this publish.                                      |
+| **/commitManifest**    | **POST**  | Send the publishing manifest                                               |
+| **/publish**           | **POST**  | Send a file to be published in this transaction (called one or more times  |
```
Comment (ian-kent) [inline README.md]: pedantic: missing a closing bracket

File: README.md [hunk 2]
```diff
@@ -1,33 +1,95 @@
...
+| **/commit**            | **POST**  | Once all the publish content has been sent begin moving the content onto the web box |
+| **/rollback**          | **POST**  | Attempt to revert the publish if something goes wrong |
+| **/transaction**       | **GET**   | Get the request transaction |
+
+#### Pre-publish steps
+For scheduled publishes we execute the _begin_ and _commitManifest_ steps slightly _ahead_ of the publish time as a 
+minor performance gain. When its time to execute the actual publish the transation exists so we only need to move the
+ content to web box. **As previously mentioned this service is earmarked for depreication. There are some fundamental
```
Comment (ian-kent) [inline README.md]: more pedantry: misspelled deprecation

File: README.md [hunk 3]
```diff
@@ -1,33 +1,95 @@
...
+  issues and perfomance is one of them.** 
+
+## Getting started
+
+### Prerequisites 
+- git
+- Java 8
+- Maven
+- [jq][3] _(not required but it's super useful for any JSON app)_
```
Comment (ian-kent) [inline README.md]: 👍

File: README.md [hunk 4]
```diff
@@ -1,33 +1,95 @@
...
+    "uriDeletes": [],
+    "errors": []
+  }
+}
+```
 
- * `EncryptedFileItemFactory`
- * `EncryptedFileItem`
- * `EncryptedDeferredOutputStream`
+You should also have a directory name the same as the transation ID in your response under your
```
Comment (ian-kent) [inline README.md]: misspelled transaction

File: default-run.sh [hunk 5]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+mkdir -p target/website
+mkdir -p target/transactions
+
+# The port of the underlying HTTP server
+export PORT=8084
```
Comment (ian-kent) [inline default-run.sh]: is it worth making this `export PORT="${PORT:-8084}"` so it can be overridden from the environment?

File: default-run.sh [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+# The path of the website content, for dev local set up this will be your Zebedee content "master" dir path.
+export content_path="<YOUR_PATH>"
```
Comment (ian-kent) [inline default-run.sh]: potentially the same for all other exports?

File: default-run.sh (hunk 5 above)
Comment (daiLlew) [reply default-run.sh]: I don't think that will work locally as at least 1 other app does that should you would get incorrect config for one of the apps.

**Pull request merged**